app/common/ipc-protocol.ts

```diff
@@ -0,0 +1,123 @@
+/**
+ * This module contains all common types for messages sent between
+ * backend and frontend IPC communication.
+ */
+import * as t from "io-ts"
+import {PathReporter} from "io-ts/lib/PathReporter"
+
+/**
+ * Runtime type to represent a request in the IPC communication.
+ */
+export const Request = t.type({
+  id: t.string,
+  method: t.string,
+  data: t.any
+})
```

Comment:
Oh I see, although the problem is that there might be responses that do not contain an `id`, that's why I pushed it into the meta section, for example, if the request is not a valid json string, it will fail when parsed and the response would not know which `id` the request had.

Reply:
Maybe the best solution is to leave the `id` as a *partial* so it might not be present?

Reply:
The empty `id` situation will be easy to handle by the renderer process (send the message to a default route). It sounds fine to me.